cmd/mtaBuild_test.go

```diff
@@ -43,34 +45,56 @@ func TestMarBuild(t *testing.T) {
 		existingFiles["package.json"] = "{\"name\": \"myName\", \"version\": \"1.2.3\"}"
 		fileUtils := MtaTestFileUtilsMock{existingFiles: existingFiles}
 
-		err := runMtaBuild(options, &cpe, &e, &fileUtils, &httpClient)
+		utils := newNpmMockUtilsBundle()
+		utils.execRunner = e
+
+		npmExecutor := npmExecutorMock{utils: utils, options: npmExecuteOptions{
+			install:            false,
+			runScripts:         []string{},
+			runOptions:         []string{},
+			defaultNpmRegistry: options.DefaultNpmRegistry,
+			sapNpmRegistry:     options.SapNpmRegistry,
+		}}
+
+		err := runMtaBuild(options, &cpe, &e, &fileUtils, &httpClient, &npmExecutor)
 
 		assert.Nil(t, err)
 
-		if assert.Len(t, e.Calls, 4) { // the second (unchecked) entry is the mta call
-			assert.Equal(t, "npm", e.Calls[1].Exec)
-			assert.Equal(t, []string{"config", "set", "registry", "https://example.org/npm"}, e.Calls[1].Params)
+		if assert.Len(t, npmExecutor.utils.execRunner.Calls, 3) { // the second (unchecked) entry is the mta call
+			assert.Equal(t, "npm", npmExecutor.utils.execRunner.Calls[1].Exec)
+			assert.Equal(t, []string{"config", "set", "registry", "https://example.org/npm"}, npmExecutor.utils.execRunner.Calls[1].Params)
```

Comment:
Could you also leave `e` in the asserts instead of exchanging for `npmExecutor.utils.execRunner`? Or are there somehow still two different exec runner instances involved?

Reply:
I have tested it again and apparently there are still two different instances.

Reply:
Ok, resolved, because for the runtime version it will be the same instance. I'm fine either way about the mocking code-paths.

Reply:
I have fixed it now to also only use one instance of the execRunner in the mocking implementation.